### ISSUE TITLE
typing in IData / IHeader

### ISSUE THREAD

**Issue by fzimmermann89** [PTB-MR/mrpro#190]
Running plain `mypy src/mrpo` gives the following errors

```
src/mrpro/data/_IHeader.py:70: error: Function "pydicom.tag.Tag" is not valid as a type  [valid-type]
src/mrpro/data/_IHeader.py:70: note: Perhaps you need "Callable[...]" or a callback protocol?
src/mrpro/data/_IHeader.py:76: error: Statement is unreachable  [unreachable]
src/mrpro/data/_IHeader.py:88: error: Function "pydicom.tag.Tag" is not valid as a type  [valid-type]
src/mrpro/data/_IHeader.py:88: note: Perhaps you need "Callable[...]" or a callback protocol?
src/mrpro/data/_IHeader.py:92: error: Function "pydicom.tag.Tag" is not valid as a type  [valid-type]
src/mrpro/data/_IHeader.py:92: note: Perhaps you need "Callable[...]" or a callback protocol?
src/mrpro/data/_IData.py:44: error: Item "None" of "DataElement | None" has no attribute "value"  [union-attr]
src/mrpro/data/_IData.py:49: error: Item "None" of "DataElement | None" has no attribute "value"  [union-attr]
src/mrpro/data/_IData.py:120: error: Function "pydicom.tag.Tag" is not valid as a type  [valid-type]
src/mrpro/data/_IData.py:120: note: Perhaps you need "Callable[...]" or a callback protocol?
src/mrpro/data/_IData.py:138: error: Argument 1 to "from_dicom_list" of "IHeader" has incompatible type "list[FileDataset]"; expected "list[Dataset]"  [arg-type]
src/mrpro/data/_IData.py:138: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
src/mrpro/data/_IData.py:138: note: Consider using "Sequence" instead, which is covariant
```

These look like actual bugs in our code. I did not check wich of our settings makes mypy in the actions ignore these issues.

@schuenke

**Comment by schuenke** [comment]
I still have no idea why this is not detected by the pre-commit mypy hook, but I agree that these are actual errors...

`"pydicom.tag.Tag" is not valid as a type  [valid-type]` makes sense, because `pydicom.tag.Tag` is a function and not a class 🙄 
The correct type-hint is `pydicom.tag.BaseTag`. I could correct this in #193. 

The last error you posted is related to #198. However, even if use `def make_unique(values: Sequence[float | None])`, we have to refactor the code so that mypy is not complaining about the `np.unique()` call in the elif case... We should take care of this in a future PR adressing #198 imo.

**Comment by fzimmermann89** [comment]
I believe all the issues got fixed in #193. Why they were not picked up by our pre-commit mypy settings is still unclear.
Will close.
